### PR TITLE
Add more tests for `group by`

### PR DESCRIPTION
**Pull request by KavinduZoysa**
## Purpose
Related to #36810

### CODE REVIEW

File: tests/jballerina-unit-test/src/test/resources/test-src/query/group_by_clause_with_list_ctr.bal [hunk 1]
```diff
@@ -1770,20 +1901,112 @@ function foo2() returns [int...] {
...
 // TODO: Add tests to empty groups
 // TODO: 2nd from clause uses seq vars of first from clause
-// TODO: multiple grouping keys
\ No newline at end of file
+// TODO: optional fiels in input
+// TODO: from var {name: namex, price1, price2} group by namex
+
+// from var {name, maths, science} in input
+// group by name
+// select {name: name, maths: avg:(maths), science: avg:(science)}
```
Comment (gimantha) [inline tests/jballerina-unit-test/src/test/resources/test-src/query/group_by_clause_with_list_ctr.bal:2012]:
```suggestion
// select {name: name, maths: avg:(maths), science: avg:(science)}

```

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java [hunk 2]
```diff
@@ -2302,9 +2312,12 @@ public void visit(BLangQueryExpr queryExpr) {
         boolean prevWithinQuery = withinQuery;
         // This can be set to true directly since it's invoked only for nested queries.
         this.withinQuery = true;
+        boolean prevAfterGroupBy = this.afterGroupBy;
```
Comment (gimantha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java:2315]: We need to have a few test to include multiple group by in same query to test these lines

Comment (KavinduZoysa) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java:2315]: Do you mean something like that,
https://github.com/ballerina-platform/ballerina-lang/pull/40287/files#diff-f33fd96d748c9aa10cdf8502de9966c2e338ef4ee796452e8c2ce5ca02d8c67fR777

Comment (KavinduZoysa) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java:2315]: I will introduce it in separate PR

**Review by gimantha — APPROVED** [review]
